src/controllers/dynakube/oneagent/daemonset/daemonset.go

```diff
@@ -222,6 +210,21 @@ func (dsInfo *builderInfo) podSpec() corev1.PodSpec {
 		Affinity:                      affinity,
 		TerminationGracePeriodSeconds: address.Of(defaultTerminationGracePeriod),
 	}
+
+	if dsInfo.dynakube.NeedsOneAgentReadinessProbe() {
+		podSpec.Containers[0].ReadinessProbe = &corev1.Probe{
+			ProbeHandler: corev1.ProbeHandler{
+				Exec: &corev1.ExecAction{
+					Command: dsInfo.dynakube.Status.OneAgent.Healthcheck.Test,
```

Comment:
Problem with this code:
Healthcheck - can be nil so that it will panic on line 214 `dsInfo.dynakube.NeedsOneAgentReadinessProbe()`
If you write such a negative test you'll catch this case

Reply:
Fixed!
`NeedsOneAgentReadinessProbe()` now only checks if the pointer Healthcheck != nil, which makes this part of the code nil safe.